src/iris/api.py

```diff
@@ -3048,6 +3079,66 @@ def on_get(self, req, resp, username):
         resp.body = ujson.dumps(user_data)
 
 
+class UserSettings(object):
+    allow_read_no_auth = False
+    enforce_user = True
+
+    def __init__(self, supported_timezones):
+        self.supported_timezones = supported_timezones
+
+    def on_get(self, req, resp, username):
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor()
+        cursor.execute(get_username_settings_query, req.context['username'])
+        settings = dict(cursor)
+        cursor.close()
+        connection.close()
+
+        resp.body = ujson.dumps(settings)
+
+    def on_put(self, req, resp, username):
+        try:
+            data = ujson.loads(req.context['body'])
+        except ValueError:
+            raise HTTPBadRequest('Invalid json in post body')
+
+        # Add more keys here if we eventually need more settings
+        allowed_keys = ('timezone', )
+
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor()
+
+        for name, value in data.iteritems():
+            if name not in allowed_keys:
```

Comment:
With the "name == 'timezone'" logic below we are essentially checking if this is the string timezone twice on every iteration. Not the end of the world, but there are more efficient ways to validate the settings.